pkg/dataobj/predicate.go

```diff
@@ -62,13 +65,15 @@ type (
 	// predicates.
 	LabelFilterPredicate struct {
 		Name string
+		Desc string // Description of the filter for debugging.
 		Keep func(name, value string) bool
 	}
 
 	// A LogMessageFilterPredicate is a [LogsPredicate] that requires the log message
 	// of the entry to pass a Keep function.
 	LogMessageFilterPredicate struct {
 		Keep func(line []byte) bool
+		Desc string // Description of the filter for debugging.
```

Comment:
nit: for consistency with the other function predicates, should Desc be before Keep here?